velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.h

```diff
@@ -33,6 +34,33 @@ void finalizeS3();
 
 class S3Config;
 
+struct S3UploadManager {
```

Comment:
We could simplify the struct as follows:
```
struct AsyncUploadInfo {
  std::optional<int32_t> partUploadSize{10485760};   
  std::optional<int32_t> writeFileSemaphoreNum{4};
  std::optional<int32_t> uploadThreads{16};
};
```

This AsyncUploadInfo instance can be created only when `uploadPartAsyncEnabled` is true, and then passed to `S3WriteFile`. Additionally, we can defer the creation of `uploadThreadPool_` in `S3WriteFile` until it is actually needed.